### ISSUE TITLE
Image inspect command failed error - linux/amd64

### ISSUE THREAD

**Issue by lambchr** [mthalman/docker-bump-action#29]
Hi,

I ran into this error whilst using the `docker-bump-action` tool but it looks like an error with dredge (or potentially something in my image is not supported) but let me know if I should move this issue to the `docker-bump-action` repo

```
Exception: Command failed with exit code 1: dredge image inspect
<PRIVATE-ECR-REGISTRY>/<REPO>:latest --os linux
--arch amd64
Invoke: dredge image inspect <PRIVATE-ECR-REGISTRY>/<REPO>:latest --os linux --arch amd64
Result: 
Exception: /home/runner/work/_temp/27caaad6-da85-4a8f-b320-c207c5f27823.ps1:39
Line |
  39 |    throw "command failed"
     |    ~~~~~~~~~~~~~~~~~~~~~~
     | command failed
Error: Process completed with exit code 1.
```

I can't share the private image with you but it is pretty simple so I can share the Dockerfile:

```
FROM ghcr.io/actions/actions-runner:latest

USER root

RUN apt-get update && \
    apt-get upgrade -y && \
    apt-get install -y git curl unzip

USER runner
```

and how we push to the ECR, so you can recreate it:

```
docker buildx build --platform linux/amd64,linux/arm64 --push -t <PRIVATE-ECR-REGISTRY>/<REPO>:latest .
```

Any help would be appreciated :)

**Comment by mthalman** [comment]
(Moved this to https://github.com/mthalman/docker-bump-action).

You're getting this error because docker-bump-action doesn't support private registries. There's no option to configure authentication.

**Comment by mthalman** [comment]
@lambchr - Can you tell me more about your use case? I've actually been wondering whether to continue maintaining this project or abandoning it. I feel like using https://github.com/renovatebot/renovate is a better way to go. Have you explored Renovate at all?

**Comment by lambchr** [comment]
> (Moved this to https://github.com/mthalman/docker-bump-action).
> 
> You're getting this error because docker-bump-action doesn't support private registries. There's no option to configure authentication.

So I am using the following two actions before the `docker-bump-action` to configure the correct credentials for my private ECR repo. Would the docker bump action not pick up the credentials created by these steps?

```
      - name: Configure AWS credentials
        uses: aws-actions/configure-aws-credentials@v4
        with:
          audience: <AUD>
          role-to-assume: <ROLE_ARN>
          aws-region: <REGION>

      - name: Login to Amazon ECR
        id: login-ecr
        uses: aws-actions/amazon-ecr-login@v2
```

> @lambchr - Can you tell me more about your use case? I've actually been wondering whether to continue maintaining this project or abandoning it. I feel like using https://github.com/renovatebot/renovate is a better way to go. Have you explored Renovate at all?

Sure, I have the above mentioned simple image, but the base image is updated very frequently. It also deprecates all older images every 30 days or so, so we wanted to avoid manually having to push new images every month. I was hoping to use this action to figure out when the push build needed to be run. I have had a quick look at renovate and I think it will satisfy that criteria, so if you don't want to continue supporting this action then I will take a look at that tool. Let me know what you decide :)

**Comment by mthalman** [comment]
> Would the docker bump action not pick up the credentials created by these steps?

No, because the tool runs within a container. It doesn't have the context of your login.

> I have had a quick look at renovate and I think it will satisfy that criteria, so if you don't want to continue supporting this action then I will take a look at that tool. Let me know what you decide :)

That would be my recommendation. Renovate works a bit differently because it generates PRs rather than directly invoking workflows. So the PRs need to have some sort of diff to be relevant in which case your Dockerfile would reference more specific tags or digest values. But it handles that for you. And I know from their docs that they do support authentication to private registries. I actually migrated to use Renovate for one of my repos and you can see an example of it updating a Dockerfile: https://github.com/mthalman/dredge/pull/116.

**Comment by lambchr** [comment]
> > Would the docker bump action not pick up the credentials created by these steps?
> 
> No, because the tool runs within a container. It doesn't have the context of your login.
> 
> > I have had a quick look at renovate and I think it will satisfy that criteria, so if you don't want to continue supporting this action then I will take a look at that tool. Let me know what you decide :)
> 
> That would be my recommendation. Renovate works a bit differently because it generates PRs rather than directly invoking workflows. So the PRs need to have some sort of diff to be relevant in which case your Dockerfile would reference more specific tags or digest values. But it handles that for you. And I know from their docs that they do support authentication to private registries. I actually migrated to use Renovate for one of my repos and you can see an example of it updating a Dockerfile: [mthalman/dredge#116](https://github.com/mthalman/dredge/pull/116).

Thanks, I'll take a look at your example :)
